Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -418,18 +420,50 @@ protected virtual void OnDrawGizmosSelected()
                     break;
             }
         }
-    
 
-        public void OnValidate()
+        private void Awake()
+        {
+#if UNITY_EDITOR
+            if (!EditorApplication.isPlaying)
+            {
+                if (assetVersion != CurrentAssetVersion)
+                {
+                    Undo.RecordObject(this, "version patching");
+                    PerformVersionPatching();
+                }
+            }
+#endif
+        }
+
+#region asset version migration
+#if UNITY_EDITOR
+        private const int CurrentAssetVersion = 1;
+
+        [SerializeField]
+        [HideInInspector]
+        private int assetVersion = 0;
+
+        private void PerformVersionPatching()
         {
-            if (Application.isPlaying)
-            {   // Don't validate during play mode
-                return;
+            if (assetVersion == 0)
+            {
+                Debug.Log("Upgrade GridObjectCollection on " + GetUserFriendlyName() + " from version 0 to version 1 for MRTK 2.2 release. Please save scene / prefab.");
+                // Migrate from version 0 to version 1
+                UpgradeAssetToVersion1();
```

Comment:
Nit: If you ever think that these have a chance to be chained together (i.e. someone upgrades from 2.1.0 to 2.4.0), then it might make sense to have UpgradeAssetToVersion1() directly set assetVersion = 1 (and then have the other  checks come after, so that they will incrementally be upgraded). Realistically this shouldn't matter here (because I don't think there's going to be another change here that affects row/cols in a way that would depend on this upgrade code to have run.